src/app/utils/performance-tests.spec.ts

```diff
@@ -17,85 +17,149 @@
+      "Loading children of schools",
+      schools
     );
-    await session.getSyncState().waitForChangeTo(SyncState.COMPLETED);
-    syncTimer.stop();
-    console.log("sync time", syncTimer.getDuration());
-
-    // delete index views from previous test runs; comment this to test queries on existing indices
-    // await deleteAllIndexedDB(db => db.includes("mrview"));
+  });
 
-    const childrenService = TestBed.inject<ChildrenService>(ChildrenService);
-    const indexTimer = new Timer(true);
-    await childrenService.createDatabaseIndices();
-    indexTimer.stop();
-    console.log("indexing time", indexTimer.getDuration());
+  it("load children with school info", async () => {
+    const childrenService = TestBed.inject(ChildrenService);
+    await mockDatabase.getIndexCreationPromises();
+    await comparePerformance(
+      () => childrenService.getChildren().toPromise(),
+      () => childrenService.getChildrenImproved(),
+      "ChildrenService getChildren"
+    );
+  });
 
-    expect(indexTimer.getDuration()).toBe(0); // display indexing time as failed assertion; see console for details
+  it("init activities in componentn", async () => {
+    const fixture = TestBed.createComponent(RollCallSetupComponent);
+    const rollCallSetup = fixture.componentInstance;
+    await mockDatabase.getIndexCreationPromises();
+    await comparePerformance(
+      () => rollCallSetup.loadActivities(),
+      () => rollCallSetup.loadActivitiesImproved(),
+      "RollCallSetupComponent loadActivities"
+    );
   });
 });
 
+async function comparePerformance<V, R>(
+  currentFunction: (val?: V) => Promise<R>,
+  improvedFunction: (val?: V) => Promise<R>,
+  description: string,
+  input?: V[]
+) {
+  const diffs: number[] = [];
+  if (input) {
+    for (const el of input) {
+      const diff = await getExecutionDiff(
+        () => currentFunction(el),
+        () => improvedFunction(el)
+      );
+      diffs.push(diff);
+    }
+    const avgDiff = diffs.reduce((sum, cur) => sum + cur, 0) / diffs.length;
+    fail("<" + description + "> Average improvement: " + avgDiff + "ms");
+  } else {
+    const diff = await getExecutionDiff(currentFunction, improvedFunction);
+    fail("<" + description + "> Execution time improvement " + diff + "ms");
+  }
+}
+
+async function getExecutionDiff<R>(
+  currentFunction: () => Promise<R>,
+  improvedFunction: () => Promise<R>
+): Promise<number> {
+  const currentTimer = new Timer();
+  const currentResult = await currentFunction();
+  const currentDuration = currentTimer.getDuration();
+  const improvedTimer = new Timer();
+  const improvedResult = await improvedFunction();
+  const improvedDuration = improvedTimer.getDuration();
+  expect(improvedResult).toEqual(
+    currentResult,
+    "current " +
+      JSON.stringify(currentResult) +
+      " improved " +
+      JSON.stringify(improvedResult)
+  );
+  return currentDuration - improvedDuration;
+}
+
```

Comment:
@TheSlimvReal  let's take this improved test setup into master and discard the other parts for now?

This PR is becoming more and more left behind and had some performance problems, so the load changes are not mergable at the moment.